### PR TITLE
APPS/IO-DEMO: Fix random shuffle of servers on Client

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Fix random shuffle of servers on Client.

## Why ?

1. To not do shuffle of servers prior setting `srand()`.
2. To do shuffle of servers in case of AM and (chunk_size < max_data_size).

## How ?

1. Move setting `srand()` in `adjust_opts()`.
2. Remove return in case of AM and (chunk_size < max_data_size).

### CODE REVIEW

**Comment by dmitrygx** [comment]
@brminich @evgeny-leksikov could you review pls?

File: test/apps/iodemo/io_demo.cc [hunk 1]
```diff
@@ -2138,11 +2138,13 @@ static void adjust_opts(options_t *test_opts) {
...
-        return;
+    } else {
+        test_opts->chunk_size = std::min(test_opts->chunk_size,
+                                         test_opts->max_data_size);
     }
 
-    test_opts->chunk_size = std::min(test_opts->chunk_size,
-                                     test_opts->max_data_size);
+    IoDemoRandom::srand(test_opts->random_seed);
+    LOG << "random seed: " << test_opts->random_seed;
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc:2147]: i think better move random_shuffle and the rest of the code here to do_client

Comment (dmitrygx) [reply test/apps/iodemo/io_demo.cc:2147]: done

**Review by brminich — APPROVED** [review]

File: test/apps/iodemo/io_demo.cc (hunk 1 above)
Comment (yosefe) [reply test/apps/iodemo/io_demo.cc:2147]: but also return srand() in there as well.

Comment (dmitrygx) [reply test/apps/iodemo/io_demo.cc]: > but also return srand() in there as well.

done

**Review by yosefe — APPROVED** [review]

**Pull request merged**